File: prototypes/python3/scripts/01_random_frame.py
## =========================================================================
## @author Leonardo Florez-Valencia (florez-l@javeriana.edu.co)
## =========================================================================

import math, sys
sys.path.append( '..' )
import PUJ.CGraf

# -- Input arguments
size = ( 100, 100 )
color_depth = 'rgb'
fname = 'frame.ppm'
if len( sys.argv ) > 2:
  size = ( int( sys.argv[ 1 ] ), int( sys.argv[ 2 ] ) )
# end if
if len( sys.argv ) > 3:
  color_depth = sys.argv[ 3 ]
# end if
if len( sys.argv ) > 4:
  fname = sys.argv[ 4 ]
# end if

# -- Create and save a random frame
frame = PUJ.CGraf.FrameBuffer( size, color_depth )
frame.save_as_netpbm( fname )

## eof - $RCSfile$

File: prototypes/cxx/executables/01_random_frame.cxx
// =========================================================================
// @author Leonardo Florez-Valencia (florez-l@javeriana.edu.co)
// =========================================================================

#include <algorithm>
#include <iostream>
#include <string>
#include <PUJ/CGraf/FrameBuffer.h>

int main( int argc, char** argv )
{
  unsigned short w = 100, h = 100;
  PUJ::CGraf::FrameBuffer::ColorDeep depth = PUJ::CGraf::FrameBuffer::RGB;
  std::string fname = "frame.ppm";

  if( argc > 1 ) w = std::atoi( argv[ 1 ] );
  if( argc > 2 ) h = std::atoi( argv[ 2 ] );
  if( argc > 3 )
  {
    std::string d = argv[ 3 ];
    std::transform( d.begin( ), d.end( ), d.begin( ), ::toupper );

    if     ( d == "BINARY" ) depth = PUJ::CGraf::FrameBuffer::BINARY;
    else if( d == "GRAY" )   depth = PUJ::CGraf::FrameBuffer::GRAY;
    else                     depth = PUJ::CGraf::FrameBuffer::RGB;
  } // end if
  if( argc > 4 ) fname = argv[ 4 ];

  PUJ::CGraf::FrameBuffer frame( w, h, depth );
  frame.save_as_netpbm( fname );

  return( EXIT_SUCCESS );
}

// eof - $RCSfile$
